editor/src/components/inspector/resize-to-fit-control.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { FlexRow, Icn, Tooltip } from '../../uuiui'
+import { EditorAction, EditorDispatch } from '../editor/action-types'
+import { useDispatch } from '../editor/store/dispatch-context'
+import { Substores, useEditorState } from '../editor/store/store-hook'
+import { setPropHugStrategies } from './inspector-strategies/inspector-strategies'
+import { executeFirstApplicableStrategy } from './inspector-strategies/inspector-strategy'
+
+export const ResizeToFitControlTestId = 'ResizeToFitControlTestId'
+
+interface ResizeToFitControlProps {}
+
+export const ResizeToFitControl = React.memo<ResizeToFitControlProps>(() => {
+  const dispatch = useDispatch()
+  const selectedViews = useEditorState(
+    Substores.selectedViews,
+    (store) => store.editor.selectedViews,
+    'ResizeToFitControl selectedViews',
+  )
+
+  const metadata = useEditorState(
+    Substores.metadata,
+    (store) => store.editor.jsxMetadata,
+    'ResizeToFitControl metadata',
+  )
+
+  const onMouseDown = React.useCallback(() => {
+    let batchedActions: Array<EditorAction> = []
+    const batchedDispatch: EditorDispatch = (actions) => {
+      batchedActions.push(...actions)
+    }
+    executeFirstApplicableStrategy(
+      batchedDispatch,
+      metadata,
+      selectedViews,
+      setPropHugStrategies('horizontal'),
+    )
+    executeFirstApplicableStrategy(
+      batchedDispatch,
+      metadata,
+      selectedViews,
+      setPropHugStrategies('vertical'),
+    )
+    if (batchedActions.length > 0) {
+      dispatch(batchedActions)
+    }
```

Comment:
This is fine as it stands, but if we start using this pattern more often it might make more sense for the `executeFirstApplicableStrategy` to return an array of actions to dispatch instead.

Reply:
we already have a similar function, `commandsForFirstApplicableStrategy`, I'll update the PR to use that

Reply:
(updated in https://github.com/concrete-utopia/utopia/pull/3154/commits/7aa95563d3493e736fdeaf1b1859da9a6b7bf59e)